### DISCUSSION TITLE
Query json array fields

### DISCUSSION THREAD

**Discussion by ostadomid** [pocketbase/pocketbase#1683] [Q&A] [answered]
Considering below json field in a collection

```
sellers: [
  {name: "bob", email:"bob@company.org"},
  {name: "joe", email:"joe@company.org"},
  {name: "james", email:"james@company.org"},
]

```
This filter  **sellers.0.name ~ "o"** will return all rows where **sellers** first member  contains "o" in   **name**  field

Is there a way to expand filter to not just only the first member but any index in array, something like this: 

**sellers.*.name ~ "o"** OR  **sellers.#.name ~ "o"**

**Comment by ganigeorgiev** [comment, marked as answer]
No, this is not supported at the moment because the SQLite [`json_extract`](https://www.sqlite.org/json1.html#jex) doesn't work with wildcard paths. There is workaround using `json_each` and `json_tree` but it comes with some drawbacks (depending on the json structure and nesting depth this could also lead to very poor performance) and for now support for this is not planned.

Depending on your use case, it may be better to convert the `json` field to a "sellers" collection and/or point to it (or from it) with a `relation` field (single or multiple). This way you'll also have the option for data validations and cascade operations. The relation picker ui was recently refactored in the develop branch and soon (most likely next week) the changes will be available with the next v0.12.0 release.
